packages/cli/src/public-api/v1/handlers/workflows/__tests__/workflows.service.test.ts

```diff
@@ -0,0 +1,36 @@
+import { mockInstance } from '@n8n/backend-test-utils';
+import { WorkflowEntity, WorkflowRepository } from '@n8n/db';
+
+import { updateWorkflow } from '../workflows.service';
+
+describe('updateWorkflow', () => {
+	const workflowRepository = mockInstance(WorkflowRepository);
+
+	beforeEach(() => {
+		jest.clearAllMocks();
+	});
+
+	it('should update workflow while preserving existing settings', async () => {
```

Comment:
Do we want an additional test for when no settings are passed in `updateData`?

Reply:
Done, thanks